File: C++/redistribution_characters_to_make_all_strings_equal.cpp
// no. 1897
#include <map>
#include <string>
using namespace std;

class Solution {
public:
  bool makeEqual(vector<string> &words) {
    map<char, int> char_map;
    for (int i = 0; i < words.size(); i++) {
      for (int j = 0; j < words[i].length(); j++) {
        char_map[words[i][j]]++;
      }
    }
    for (map<char, int>::iterator it = char_map.begin(); it != char_map.end();
         it++) {
      if (it->second % words.size()) {
        return false;
      }
    }
    return true;
  }
};

File: python/redistribution_characters_to_make_all_strings_equal.py
#no. 1897

class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        char_dict = {}
        for w in words:
            for c in w:
                char_dict[c] = char_dict.get(c, 0) + 1
        for c in char_dict:
            if char_dict[c] % len(words):
                return False
        return True

